From 19a7e9d3f1d8b1fef652c88526ab52193b793432 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Fri, 6 Mar 2026 07:20:06 +0100
Subject: [PATCH 3/5] dts/aspeed: aspeed-bmc-ibm-system1: Fix PCIe slots

Currently the PCIe sensors readings aren't properly working. Update the
devicetree to resolve this problem.

* Add aliases for PCIe slots 1-8.
* Move PCIe slots 9-16 after PCIe slots 1-8.
  Keep existing non-PCIe bus numbers to not break user-space applications
  hardcoding I2C bus numbers.
* Add FRU EEPROMs
  Every PCIe slot is expected to have one specialized PCIe addon card
  that is equiped with a FRU EEPROM. The slots are non hot-plugable
  and all slots are always expected to be seated by such cards.

Upstream-Status: Pending
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 .../dts/aspeed/aspeed-bmc-ibm-system1.dts     | 106 ++++++++++++++++--
 1 file changed, 98 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-ibm-system1.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-ibm-system1.dts
index 28662a66de17..002b6b679872 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-ibm-system1.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-ibm-system1.dts
@@ -12,14 +12,6 @@ / {
 	compatible = "ibm,system1-bmc", "aspeed,ast2600";
 
 	aliases {
-		i2c16 = &i2c8mux1chn0;
-		i2c17 = &i2c8mux1chn1;
-		i2c18 = &i2c8mux1chn2;
-		i2c19 = &i2c8mux1chn3;
-		i2c20 = &i2c8mux1chn4;
-		i2c21 = &i2c8mux1chn5;
-		i2c22 = &i2c8mux1chn6;
-		i2c23 = &i2c8mux1chn7;
 		i2c24 = &i2c3mux0chn0;
 		i2c25 = &i2c3mux0chn1;
 		i2c26 = &i2c3mux0chn2;
@@ -68,6 +60,24 @@ aliases {
 		i2c69 = &i2c15mux0chn5;
 		i2c70 = &i2c15mux0chn6;
 		i2c71 = &i2c15mux0chn7;
+		// PCIe SLOTs 1-8
+		i2c72 = &i2c15mux1chn0;
+		i2c73 = &i2c15mux1chn1;
+		i2c74 = &i2c15mux1chn2;
+		i2c75 = &i2c15mux1chn3;
+		i2c76 = &i2c15mux1chn4;
+		i2c77 = &i2c15mux1chn5;
+		i2c78 = &i2c15mux1chn6;
+		i2c79 = &i2c15mux1chn7;
+                // PCIe SLOTs 9-16
+		i2c80 = &i2c8mux1chn0;
+		i2c81 = &i2c8mux1chn1;
+		i2c82 = &i2c8mux1chn2;
+		i2c83 = &i2c8mux1chn3;
+		i2c84 = &i2c8mux1chn4;
+		i2c85 = &i2c8mux1chn5;
+		i2c86 = &i2c8mux1chn6;
+		i2c87 = &i2c8mux1chn7;
 	};
 
 	chosen {
@@ -1257,48 +1267,88 @@ i2c8mux1chn0: i2c@0 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <0>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn1: i2c@1 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <1>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn2: i2c@2 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <2>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn3: i2c@3 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <3>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn4: i2c@4 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <4>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn5: i2c@5 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <5>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn6: i2c@6 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <6>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c8mux1chn7: i2c@7 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <7>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 			};
 		};
@@ -1702,48 +1752,88 @@ i2c15mux1chn0: i2c@0 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <0>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn1: i2c@1 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <1>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn2: i2c@2 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <2>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn3: i2c@3 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <3>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn4: i2c@4 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <4>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn5: i2c@5 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <5>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn6: i2c@6 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <6>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 
 				i2c15mux1chn7: i2c@7 {
 					#address-cells = <1>;
 					#size-cells = <0>;
 					reg = <7>;
+
+					eeprom@51 {
+						compatible = "atmel,24c32";
+						reg = <0x51>;
+					};
 				};
 			};
 		};
-- 
2.47.1

